wallet/src/Pos/Wallet/Web/Pending/Submission.hs

```diff
@@ -166,9 +174,63 @@ submitAndSavePtx submitTx PtxSubmissionHandlers{..} ptx@PendingTx{..} = do
         -- while transaction creation failed, due to protocol error or bug,
         -- then we better not remove this pending transaction
         void $ usingPtxCoords removeOnlyCreatingPtx ptx
+
     badInitPtxCondition = InternalError "Expected PtxCreating as initial pending condition"
 
     reportSubmitted ack =
         logDebug $
         sformat ("submitAndSavePtx: transaction submitted with confirmation?: "
                 %build) ack
+
+
+-- As simple and as clear as I could.
+submitAndSavePtxStates
+    :: TxSubmissionMode ctx m
+    => PendingTx
+    -> Timestamp
+    -> ((TxId, TxAux) -> m TxSubmissionResult)
+    -> m TxSubmissionResult
+submitAndSavePtxStates ptx@PendingTx{..} now mSaveTx =
+
+    case _ptxCond of
+
+      PtxApplying poolInfo -> do
+          let Just creationTime = poolInfo ^. thTimestamp
+          if diffTimestamp now creationTime > hour 1
```

Comment:
Same for this condition. Previously when this is `False`, second branch of `case` was executed. Now you assign special `TxStillApplying` state which corresponds to doing nothing.

Reply:
Right, but won't the transaction still be in the original state `PtxApplying` and will be picked up by the re-submission mechanism, right?

Reply:
Yes, but you still will do nothing if condition is `False`, because you treat `StillApplying` differently. Or not?

Reply:
Ah, I see. You are correct, nice catch. 
Thanks for catching this, I would have been in trouble if this went through :sweat_smile: